src/server/game/Combat/CombatManager.cpp

```diff
@@ -17,6 +17,33 @@
 
 #include "CombatManager.h"
 #include "Unit.h"
+#include "Player.h"
+
+/*static*/ bool CombatManager::IsCombatValid(Unit const* a, Unit const* b)
+{
+    // Checks combat validity before initial reference creation.
+    // For the combat to be valid...
+    // ...the two units need to be different
+    if (a == b)
+        return false;
+    // ...the two units need to be in the world
+    if (!a->IsInWorld() || !b->IsInWorld())
+        return false;
+    // ...the two units need to both be alive
+    if (!a->IsAlive() || !b->IsAlive())
+        return false;
+    // ...the two units need to be on the same map
+    if (a->GetMap() != b->GetMap())
+        return false;
+    // ...the two units need to be in the same phase
```

Comment:
Apparently, yes. Vali adds should continue to attack you while you're not in their phase? That sounds like a bug to me (would be completely unique behavior).